### PR TITLE
test: add the unit test for computing the `domain_size`

### LINKED ISSUE #1335: Add a test checking the domain size when building a circuit with gates using same table.
Approving, but would have been nice to have a test, even it looks fine regarding the code. Opening a follow-up issue for a 10 min task.

_Originally posted by @dannywillems in https://github.com/o1-labs/proof-systems/pull/1334#pullrequestreview-1734526395_
           
 Before this PR, it was duplicating the domain if two gates were using the same table.

### PR DESCRIPTION
**Pull request by duguorong009**
## Description
Add the unit test for checking the `domain_size` of circuit, when the gates use the same table

## Related issues
- Resolve #1335 

## Changes
- add the `test_lookup_domain_size_computation` in `kimchi/src/circuits/constraints.rs`

### CODE REVIEW

**Review by dannywillems — COMMENTED** [review]
Thanks for the contribution! Really appreciated!
The code looks good and clean, good job!

However, I do not think it does fulfil all the requirements to close the issue. The idea of the issue was to verify that if a circuit uses the gate [Foreign Field Multiplication](https://o1-labs.github.io/proof-systems/kimchi/foreign_field_mul.html) alongside the gate "rangecheck", then, the table "rangecheck" will be added only once in the domain size computation. You can see that the foreign field multiplication gate does require performing a range check.

The tests you are suggested are still correct. I would leave this review as a comment, for now, in case you would like to add additional tests based on my comment above. Happy to give more guidance, if needed.

**Comment by duguorong009** [comment]
> Thanks for the contribution! Really appreciated! The code looks good and clean, good job!
> 
> However, I do not think it does fulfil all the requirements to close the issue. The idea of the issue was to verify that if a circuit uses the gate [Foreign Field Multiplication](https://o1-labs.github.io/proof-systems/kimchi/foreign_field_mul.html) alongside the gate "rangecheck", then, the table "rangecheck" will be added only once in the domain size computation. You can see that the foreign field multiplication gate does require performing a range check.
> 
> The tests you are suggested are still correct. I would leave this review as a comment, for now, in case you would like to add additional tests based on my comment above. Happy to give more guidance, if needed.

Thanks for review! @dannywillems 
How about this simple addition? https://github.com/o1-labs/proof-systems/pull/2369/commits/7c22692fa36f87acfa611d6a4b9c64129efd9a46

File: kimchi/src/circuits/constraints.rs [hunk 1]
```diff
@@ -961,4 +962,54 @@ pub mod tests {
...
+            (
+                (0, 0),
+                8192, /* 8192 > 0 * 0 + 1 * 4096 + 1 * 256 + 1 + zk_row */
+            ),
+            (
+                (5, 100),
+                8192, /* 8192 > 5 * 100 + 1 * 4096 + 1 * 256 + 1 + zk_row */
+            ),
+        ];
+        for ((number_of_table_ids, size), expected_domain_size) in data.into_iter() {
```
Comment (dannywillems) [inline kimchi/src/circuits/constraints.rs:997]: Nit: personal preference to use `data.into_iter().for_each()` for a functional approach.

**Review by dannywillems — APPROVED** [review]
LGTM, thank you! I approve.

File: kimchi/src/circuits/constraints.rs (hunk 1 above)
Comment (duguorong009) [reply kimchi/src/circuits/constraints.rs]: Done https://github.com/o1-labs/proof-systems/pull/2369/commits/32637674c1975382ae82179db22b9324a11f8c43

**Pull request merged**